core/src/main/java/org/apache/sling/testing/mock/osgi/MapMergeUtil.java

```diff
@@ -98,6 +104,9 @@ private MapMergeUtil() {
             mergedProperties.putAll(properties);
         }
 
+        // add non overwritable properties
+        mergedProperties.put(ComponentConstants.COMPONENT_NAME, targetClass.getName());
```

Comment:
maybe a better place would be `MockServiceRegistration#updateProperties` where other "implicit" properties are set already? this would ensure all code paths are affected regardless by which way the component was created when it is registered.

Reply:
This is not a framework services property, but one which is exclusively used for DS components!

Reply:
ah, was not aware of this. this goes a bit along with the discussion above - if we want just apply the DS magic everyhwere or try to be super strict and make a distinction (which may not be obvious for most of the mock users).

Reply:
Treating regular services like DS components is problematic. Let us fix this and maybe clarify the javadoc of registerService methods. But if this MockOsgi.registerService() behaves differently than BundleContext.registerService(...) that is definitely not expected and should be fixed.